src/dotnet-interactive-vscode/src/stdioKernelTransport.ts

```diff
@@ -18,41 +19,100 @@ import { LineReader } from './lineReader';
 import { parse, stringify } from './utilities';
 
 export class StdioKernelTransport {
-    private childProcess: cp.ChildProcessWithoutNullStreams;
+    private childProcess: cp.ChildProcessWithoutNullStreams | null;
     private lineReader: LineReader;
     private readyPromise: Promise<void>;
     private subscribers: Array<KernelEventEnvelopeObserver> = [];
+    public httpPort: Number;
 
     constructor(processStart: ProcessStart, private diagnosticChannel: ReportChannel) {
         // prepare root event handler
         this.lineReader = new LineReader();
         this.lineReader.subscribe(line => this.handleLine(line));
+        this.childProcess = null;
+        this.httpPort = 0;
 
         // prepare one-time ready event
-        this.readyPromise = new Promise<void>((resolve, reject) => {
+        this.readyPromise = new Promise<void>(async (resolve, reject) => {
+
+            let args = await this.configureHttpArgs(processStart.args);
+            // launch the process
+            let childProcess = cp.spawn(processStart.command, args, { cwd: processStart.workingDirectory });
+            let pid = childProcess.pid;
+
+            this.childProcess = childProcess;
+            this.diagnosticChannel.appendLine(`Kernel started with pid ${childProcess.pid}.`);
+
+            childProcess.on('exit', (code: number, signal: string) => {
+
+                let message = `Kernel pid ${childProcess.pid} ended`;
+                let messageCodeSuffix = (code && code !== 0)
+                    ? ` with code ${code}`
+                    : '';
+                let messageSignalSuffix = signal
+                    ? ` with signal ${signal}`
+                    : '';
+                this.diagnosticChannel.appendLine(message + messageCodeSuffix + messageSignalSuffix);
+            });
+
+            childProcess.stdout.on('data', data => this.lineReader.onData(data));
+            childProcess.stderr.on('data', data => this.diagnosticChannel.appendLine(`kernel (${pid}) stderr: ${data.toString('utf-8')}`));
+
+
             const readySubscriber = this.subscribeToKernelEvents(eventEnvelope => {
                 if (eventEnvelope.eventType === KernelReadyType) {
                     readySubscriber.dispose();
                     resolve();
                 }
             });
         });
+    }
+
+    private async configureHttpArgs(args: string[]): Promise<string[]> {
+        let newArgs = [...args];
+        let index = newArgs.indexOf("--http-port");
+        if (index < 0) {
+            index = newArgs.indexOf("--http-port-range");
+            this.httpPort = await this.findFreePort();
+
+            if (index > 0) {
+                this.diagnosticChannel.appendLine("the --http-port-range command option is not supported in vscode extension, remove if from settings. The kernel will start with --http-port option isntead.");
```

Comment:
```suggestion
                this.diagnosticChannel.appendLine("The --http-port-range option is not supported in the VS Code extension. Please use --http-port instead.");
```

Reply:
Should we error out instead? This seems like it will work nondeterministically, and if the user isn't looking at the console, they won't see the warning.

Reply:
this will be  removed later